packages/headless/src/index.ts

```diff
@@ -125,8 +126,9 @@ export default async (opts: HeadlessOptions) => {
     throw new Error(`Headless installation requires a ${WANTED_LOCKFILE} file`)
   }
 
-  const rootModulesDir = await realpathMissing(path.join(lockfileDir, 'node_modules'))
-  const virtualStoreDir = pathAbsolute(opts.virtualStoreDir ?? 'node_modules/.pnpm', lockfileDir)
+  const relativeModulesDir = opts.modulesDir ?? 'node_modules'
+  const rootModulesDir = await realpathMissing(path.join(lockfileDir, relativeModulesDir))
+  const virtualStoreDir = pathAbsolute(opts.virtualStoreDir ?? path.join(relativeModulesDir, '.pnpm'), lockfileDir)
```

Comment:
I'm curious, how does `pathAbsolute` differs from `path.resolve`?

Reply:
pathAbsolute can resolve tilda (~/) in directory paths.

Reply:
That's what I figured.